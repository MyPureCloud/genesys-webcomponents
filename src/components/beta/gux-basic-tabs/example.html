<h1>gux-basic-tabs</h1>

<h2>Example 1</h2>

<gux-basic-tabs id="gux-tabs-example-1">
  <span slot="tabs">
    <gux-basic-tab tab-id="1">
      <gux-basic-tab-title>
        <span slot="title">Hello 1</span>
      </gux-basic-tab-title>
    </gux-basic-tab>
    <gux-basic-tab tab-id="2">
      <gux-basic-tab-title>
        <span slot="title">Hello 2 long long long</span>
      </gux-basic-tab-title>
    </gux-basic-tab>
    <gux-basic-tab tab-id="3">
      <gux-basic-tab-title>
        <span slot="title">Hello 3</span>
      </gux-basic-tab-title>
    </gux-basic-tab>
    <gux-basic-tab tab-id="4">
      <gux-basic-tab-title>
        <span slot="title">Hello 4</span>
      </gux-basic-tab-title>
    </gux-basic-tab>
    <gux-basic-tab tab-id="5">
      <gux-basic-tab-title>
        <span slot="title">Hello 5</span>
      </gux-basic-tab-title>
    </gux-basic-tab>
    <gux-basic-tab tab-id="6">
      <gux-basic-tab-title>
        <span slot="title">Hello 6</span>
      </gux-basic-tab-title>
    </gux-basic-tab>
    <gux-basic-tab tab-id="7">
      <gux-basic-tab-title>
        <span slot="title">Hello 7</span>
      </gux-basic-tab-title>
    </gux-basic-tab>
    <gux-basic-tab tab-id="8">
      <gux-basic-tab-title>
        <span slot="title">Hello 8</span>
      </gux-basic-tab-title>
    </gux-basic-tab>
    <gux-basic-tab tab-id="9">
      <gux-basic-tab-title>
        <span slot="title">Hello 9</span>
      </gux-basic-tab-title>
    </gux-basic-tab>
    <gux-basic-tab tab-id="10">
      <gux-basic-tab-title>
        <span slot="title">Hello 10</span>
      </gux-basic-tab-title>
    </gux-basic-tab>
  </span>
</gux-basic-tabs>

<h2>Example 2 -- With tabpanel content</h2>

<gux-basic-tabs id="gux-tabs-example-2">
  <span slot="tabs">
    <gux-basic-tab tab-id="11">
      <gux-basic-tab-title>
        <span slot="title">Hello 1</span>
      </gux-basic-tab-title>
    </gux-basic-tab>
    <gux-basic-tab tab-id="12">
      <gux-basic-tab-title>
        <span slot="title">Hello 2 long long long</span>
      </gux-basic-tab-title>
    </gux-basic-tab>
    <gux-basic-tab tab-id="13">
      <gux-basic-tab-title>
        <span slot="title">Hello 3 long long long</span>
      </gux-basic-tab-title>
    </gux-basic-tab>
  </span>
  <span slot="tabContentPanel">
    <p>
      "Sed ut perspiciatis unde omnis iste natus error sit voluptatem
      accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab
      illo inventore veritatis et quasi architecto beatae vitae dicta sunt
      explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut
      odit aut fugit, sed quia consequuntur magni dolores eos qui ratione
      voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum
      quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam
      eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat
      voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam
      corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur?
      Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam
      nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas
      nulla pariatur?"
    </p>
    <gux-button>sample button</gux-button>
  </span>
</gux-basic-tabs>

<h2>Example 5 -- only icons</h2>

<gux-basic-tabs id="gux-tabs-example-5">
  <span slot="tabs">
    <gux-basic-tab tab-id="27">
      <gux-basic-tab-icon>
        <gux-icon icon-name="trashcan" decorative="true"></gux-icon>
        <span slot="title">Trash 1 </span>
      </gux-basic-tab-icon>
    </gux-basic-tab>
    <gux-basic-tab tab-id="28">
      <gux-basic-tab-icon>
        <gux-icon icon-name="trashcan" decorative="true"></gux-icon>
        <span slot="title">Trash 2 </span>
      </gux-basic-tab-icon>
    </gux-basic-tab>
    <gux-basic-tab tab-id="29">
      <gux-basic-tab-icon>
        <gux-icon icon-name="trashcan" decorative="true"></gux-icon>
        <span slot="title">Trash 3 </span>
      </gux-basic-tab-icon>
    </gux-basic-tab>
  </span>
</gux-basic-tabs>

<h2>Example 6 icons and text</h2>

<gux-basic-tabs id="gux-tabs-example-6">
  <span slot="tabs">
    <gux-basic-tab tab-id="30">
      <gux-basic-tab-icon-title>
        <gux-icon slot="icon" icon-name="trashcan" decorative="true"></gux-icon>
        <span slot="title">Hello 1</span>
      </gux-basic-tab-icon-title>
    </gux-basic-tab>
    <gux-basic-tab tab-id="31">
      <gux-basic-tab-icon-title>
        <gux-icon slot="icon" icon-name="trashcan" decorative="true"></gux-icon>
        <span slot="title">Hello 2</span>
      </gux-basic-tab-icon-title>
    </gux-basic-tab>
    <gux-basic-tab tab-id="32">
      <gux-basic-tab-icon-title>
        <gux-icon slot="icon" icon-name="trashcan" decorative="true"></gux-icon>
        <span slot="title">Hello 3 longlonglonglong</span>
      </gux-basic-tab-icon-title>
    </gux-basic-tab>
  </span>
</gux-basic-tabs>
