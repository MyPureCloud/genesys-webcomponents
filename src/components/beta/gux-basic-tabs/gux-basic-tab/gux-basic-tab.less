@import (reference) '../../../../style/color.less';
@import (reference) '../../../../style/spacing.less';
@import (reference) '../../../../style/mixins.less';

@gux-off-tab-fill: @gux-grey-40;

// Style
gux-basic-tab {
  .tab-title {
    color: @gux-type;
  }

  .gux-basic-tab {
    display: flex;
    align-items: center;
    max-width: 160px;
    height: 40px;
    padding: 0 @spacing-medium;
    overflow: hidden;
    text-overflow: ellipsis;
    background-color: @gux-grey-90;
    border: none;

    // &.gux-show {
    //   width: 160px;
    // }

    .tab-title {
      overflow: hidden;
      font-size: 14px;
      white-space: nowrap;
      cursor: default;
      &.gux-hidden {
        display: none;
      }
    }

    &:focus-visible {
      outline: none;
      .focus-ring();
    }

    &.selected {
      border-bottom: 3px solid @gux-blue-60;
    }

    &:hover:not(.selected) {
      border-bottom: 1px solid @gux-grey-10;
    }

    &.disabled {
      .tab-title {
        color: @gux-black-50;
      }
    }
  }
}
