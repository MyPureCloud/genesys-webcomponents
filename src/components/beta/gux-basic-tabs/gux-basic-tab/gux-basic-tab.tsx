import {
  Component,
  Element,
  Event,
  EventEmitter,
  h,
  JSX,
  Prop,
  State
} from '@stencil/core';

import { eventIsFrom } from '../../../../utils/dom/event-is-from';

@Component({
  styleUrl: 'gux-basic-tab.less',
  tag: 'gux-basic-tab'
})
export class GuxBasicTab {
  private tabName: string;

  @Element()
  private root: HTMLElement;
  /**
   * unique id for the tab
   */
  @Prop() tabId: string;

  /**
   * indicates whether or not the tab is active
   */
  @Prop() active: boolean = false;

  /**
   * indicates the gux-icon to display on the left side of the tab (similar to a favicon in the browser)
   */
  @Prop() tabIconName: string;

  @Prop() iconOnly: boolean;

  @State() private showTooltip: boolean = true;

  @Event()
  private internaltabactive: EventEmitter<void>;

  private selectTab(e: MouseEvent): void {
    if (eventIsFrom('.tab-dropdown-container', e)) {
      return;
    }

    this.internaltabactive.emit();
  }

  componentWillLoad() {
    this.tabName = this.root.querySelector('[slot="title"]').innerHTML;
    this.checkForTooltipHideOrShow();
  }

  private checkForTooltipHideOrShow() {
    const clientWidth = this.root.clientWidth;
    // console.log(clientWidth, 'clientWidth')
    if (clientWidth < 113 && !this.root.querySelector('gux-basic-tab-icon')) {
      this.showTooltip = false;
    }
  }

  render(): JSX.Element {
    return [
      <button
        class={{
          'gux-basic-tab': true,
          'gux-active': this.active
        }}
        type="button"
        onClick={e => this.selectTab(e)}
        role="button"
      >
        <slot />
      </button>,
      this.renderTooltip()
    ];
  }

  private renderTooltip() {
    if (this.showTooltip) {
      return <gux-tooltip>{this.tabName}</gux-tooltip>;
    }
  }
}
