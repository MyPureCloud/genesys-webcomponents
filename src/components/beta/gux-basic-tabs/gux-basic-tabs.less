@import (reference) '../../../style/color.less';
@import (reference) '../../../style/mixins.less';

// Variables part
@gux-tabs-background: @gux-grey-60;
@gux-off-tab-fill: @gux-grey-40;

// Style
.gux-basic-tabs {
  // .gux-left {

  // }
  // .gux-full {

  // }
  // .gux-center {

  // }

  // .gux-vertical {

  // }

  // .gux-horizontal {

  // }
  display: flex;
  justify-content: flex-end;
  // align-items: flex-end;
  width: 100%;
  height: 40px;
  overflow-x: hidden;
  border-bottom: 1px solid @gux-grey-50;

  .action-button-container {
    display: flex;
    border-radius: 4px;

    .arrow-button {
      display: flex;
      align-items: center;
      justify-content: center;
      width: 28px;
      height: 40px;

      // padding: 4px 16px;

      border: none;
      // border-style: solid;
      // border-width: 1px;

      gux-icon {
        width: 10px;
        height: 10px;
      }

      &:focus-visible {
        .focus-ring();
      }
    }
  }

  .scrollable-section {
    display: flex;
    flex: 1 1 auto;
    overflow-x: auto;
    scroll-behavior: smooth;
    scrollbar-width: none;

    &::-webkit-scrollbar {
      height: 0;
    }
  }
}

// Theming
// Light
.gux-tabs-light-theme {
  background-color: @gux-grey-90;

  .action-button-container {
    background-color: @gux-off-tab-fill;

    .arrow-button {
      color: @gux-black-30;
      background-color: @gux-tabs-background;
      border-color: @gux-off-tab-fill;

      &:hover {
        color: @gux-blue-60;
      }

      &:active {
        background-color: darken(@gux-tabs-background, 5%);
      }
    }

    .add-tab {
      color: @gux-black-30;
      background-color: @gux-tabs-background;
      border-color: @gux-off-tab-fill;

      &:hover {
        color: @gux-blue-60;
      }

      &:active {
        background-color: darken(@gux-tabs-background, 5%);
      }
    }
  }

  .scrollable-section {
    &::-webkit-scrollbar-track {
      background: @gux-off-white;
    }

    &::-webkit-scrollbar-thumb {
      background: @gux-black-30;
    }

    &::-webkit-scrollbar-thumb:hover {
      background: @gux-black-30;
    }

    .add-tab {
      color: @gux-black-30;
      background-color: @gux-off-tab-fill;

      &:hover {
        color: @gux-blue-60;
      }

      &:active {
        background-color: darken(@gux-off-tab-fill, 5%);
      }
    }
  }
}

.gux-light-theme {
  .gux-basic-tabs {
    .gux-tabs-light-theme();
  }
}

.gux-basic-tabs.gux-light-theme {
  .gux-tabs-light-theme();
}

// Default
.gux-basic-tabs {
  .gux-tabs-light-theme();
}
