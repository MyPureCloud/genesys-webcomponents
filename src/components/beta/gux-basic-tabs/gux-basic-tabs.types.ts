export type GuxBasicTabsAlignment = 'left' | 'center' | 'full';

export type GuxBasicTabsOrientation = 'vertical' | 'horizontal';
