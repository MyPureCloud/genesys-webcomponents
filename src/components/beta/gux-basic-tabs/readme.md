# gux-tabs



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute     | Description                         | Type                           | Default        |
| ------------- | ------------- | ----------------------------------- | ------------------------------ | -------------- |
| `alignment`   | `alignment`   | Tab alignment                       | `"center" \| "full" \| "left"` | `'left'`       |
| `orientation` | `orientation` | Tab orientation                     | `"horizontal" \| "vertical"`   | `"horizontal"` |
| `value`       | `value`       | tabId of the currently selected tab | `string`                       | `''`           |


## Events

| Event   | Description                      | Type               |
| ------- | -------------------------------- | ------------------ |
| `input` | Triggers when a tab is selected. | `CustomEvent<any>` |


## Dependencies

### Depends on

- [gux-icon](../../stable/gux-icon)

### Graph
```mermaid
graph TD;
  gux-basic-tabs --> gux-icon
  style gux-basic-tabs fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
