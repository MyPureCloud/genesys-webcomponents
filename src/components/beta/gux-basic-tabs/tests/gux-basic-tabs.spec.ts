import { newSpecPage, SpecPage } from '@stencil/core/testing';

import { GuxBasicTabs } from '../gux-basic-tabs';
import { GuxBasicTab } from '../gux-basic-tab/gux-basic-tab';

const components = [GuxBasicTabs, GuxBasicTab];
const language = 'en';

describe('gux-basic-tabs', () => {
  let page: SpecPage;

  beforeEach(async () => {
    page = await newSpecPage({
      components,
      html: `
        <gux-basic-tabs id="interactive">
          <span slot="tabs">
            <gux-tab tab-id="1">
              <span slot="title"> Hello World </span>
            </gux-tab>

            <gux-tab tab-id="2">
              <span slot="title"> Hello World 2 </span>
            </gux-tab>
          </span>
        </gux-basic-tabs>
      `,
      language
    });
  });

  it('should build', async () => {
    expect(page.rootInstance).toBeInstanceOf(GuxBasicTabs);
  });

  it('should render', async () => {
    expect(page.root).toMatchSnapshot();
  });
});
