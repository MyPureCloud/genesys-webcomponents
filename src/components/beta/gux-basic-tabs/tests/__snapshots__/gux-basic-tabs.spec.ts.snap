// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-basic-tabs should render 1`] = `
<gux-basic-tabs id="interactive">
  <mock:shadow-root>
    <div class="gux-basic-tabs gux-horizontal gux-left">
      <div class="scroll-button-container"></div>
      <div class="scrollable-section">
        <slot name="tabs"></slot>
      </div>
      <div class="scroll-button-container"></div>
    </div>
    <div>
      <slot name="tabContentPanel"></slot>
    </div>
  </mock:shadow-root>
  <span slot="tabs">
    <gux-tab tab-id="1">
      <span slot="title">
        Hello
      </span>
    </gux-tab>
    <gux-tab tab-id="2">
      <span slot="title">
        Hello 2
      </span>
    </gux-tab>
  </span>
</gux-basic-tabs>
`;
